mantidproject/mantid
# Remove data processor interface

**Description of work.**
This PR removes the [DataProcessor](https://docs.mantidproject.org/v6.4.0/interfaces/framework/DataProcessorWidget.html) widget, which has been superseded by the [JobTreeView](https://developer.mantidproject.org/BatchWidget/API/JobTreeView.html). The DataProcessor widget has not been in use for several years and is unlikely to ever be used again.

This PR also removes a testhelpers directory which was only used by this widget.

Fixes #33643

**To test:**



Most of the changes are the removal of the DataProcessorUI module.

- Check that the removed directories make sense
- Check that the changes to CMakeLists etc. make sense. 
- Check that MantidWorkbench starts up ok.

*This does not require release notes* because **it removes unused code**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
